nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -359,26 +369,97 @@ def _get_quantization_target_points(self, model: TModel) -> OrderedDict[TargetPo
         pattern = PatternsManager.get_full_pattern_graph(backend, device)
         quantizer_setup = self._get_quantizer_setup(nncf_graph, pattern)
         self._apply_model_type_pass(self._parameters.model_type, quantizer_setup, nncf_graph)
-        for quantization_point in quantizer_setup.quantization_points.values():
+        quantization_points = list(quantizer_setup.quantization_points.values())
+        quantization_points = self._topological_sort_quantization_points(quantization_points, nncf_graph)
+        for quantization_point in quantization_points:
             if quantization_point.is_weight_quantization_point():
                 self._add_weight_quantization_target_point(quantization_point, nncf_graph)
             elif quantization_point.is_activation_quantization_point():
                 self._add_activation_quantization_target_point(quantization_point)
             else:
                 raise RuntimeError('Incorrect quantization point')
-        self._quantization_target_points_to_qconfig = collections.OrderedDict(
-            sorted(self._quantization_target_points_to_qconfig.items()))
         return self._quantization_target_points_to_qconfig
 
+    def _topological_sort_quantization_points(self, quantization_points: List[SingleConfigQuantizationPoint],
+                                              nncf_graph: NNCFGraph) -> List[SingleConfigQuantizationPoint]:
+        """
+        Sorts quantization_points based on the topological order of nodes obtained form nncf_graph.
+
+        :param quantization_points: Quantization points.
+        :param nncf_graph: Instance of NNCFgraph used to get topological sort.
+        :return: Sorted quantization_points.
+        """
+        node_names_to_pos = {node.node_name: i for i, node in enumerate(nncf_graph.topological_sort())}
+        quantization_points.sort(key=lambda point: node_names_to_pos[point.insertion_point.target_node_name])
+        return quantization_points
+
+    def _get_first_quantized_weight_nodes(self, quantization_points: List[TargetPoint],
+                                          starting_node: NNCFNode,
+                                          nncf_graph: NNCFGraph) -> Optional[List[TargetPoint]]:
+        """
+        Returns target points connected to a first visited node, which are included in quantization_points.
+        A traversal of nncf_graph is started from starting_node.
+
+        :param quantization_points: Quantization target points.
+        :param starting_node: Node from which traversing is started.
+        :param nncf_graph: Instance of NNCFGraph to traverse.
+        :return: First visited target points.
+        """
+        target_node_names_to_qp = {}
```

Comment:
Was it tested with the ignored scope & model with multiple inputs?

Reply:
If any node is included in an ignored scope, `QPS` will return the `quantizer setup` which doesn't have a quantizer on this operation. It means that if this quantizer is expected to be the first, then the function returns a followed by the quantizer will be returned.
So, I suppose there is no need to add a test covering such a case, cause we have a test on `ignored_scope`.

At the same time, I can add a unit test on synthetic `nncf_graph` and `quantization_points` to check whether the function returns correct target points, covering the case when NNCFGraph has multiple inputs and when there is no first quantized node in `quantization_points`